src/main.cpp

```diff
@@ -2080,6 +2081,50 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+// Protected by cs_main
+static VersionBitsCache versionbitscache;
+
+int32_t ComputeBlockVersion(const CBlockIndex* pindexPrev, const Consensus::Params& params)
```

Comment:
@sipa I tried implementing a unit test for `ComputeBlockVersion()` using a dummy deployment; not sure adding a dummy softfork for testing is something we want to do, but if you're interested the code for that is here:

https://github.com/sdaftuar/bitcoin/commit/b8ec634651fbc5a41afc47a768de4357983d3b4f

If you think this is worth including in some form, feel free to grab it, or I am happy to separately PR this unit test later.